Formula/p/python-matplotlib.rb

```diff
@@ -0,0 +1,67 @@
+class PythonMatplotlib < Formula
+  desc "Python library for creating static, animated, and interactive visualizations"
+  homepage "https://matplotlib.org/"
+  url "https://files.pythonhosted.org/packages/fb/ab/38a0e94cb01dacb50f06957c2bed1c83b8f9dac6618988a37b2487862944/matplotlib-3.8.2.tar.gz"
+  sha256 "01a978b871b881ee76017152f1f1a0cbf6bd5f7b8ff8c96df0df1bd57d8755a1"
+  license "PSF-2.0"
+
+  depends_on "meson" => :build
+  depends_on "meson-python" => :build
+  depends_on "ninja" => :build
+  depends_on "pkg-config" => :build
+  depends_on "pybind11" => :build
+  depends_on "python-build" => :build
+  depends_on "python-certifi" => :build
+  depends_on "python-setuptools-scm" => :build
+  depends_on "fonttools"
```

Comment:
this needs a `.pth` file. you can check that all requirements are satisfied with:

```console
$ python3.12 -m pip check
matplotlib 3.8.2 requires fonttools, which is not installed.
matplotlib 3.8.2 has requirement python-dateutil>=2.7, but you have python-dateutil 0.0.0.
```

Reply:
#154826 fixes the `python-dateutil` version

Reply:
`fonttools` will not need a `.pth` file once #154672 is merged.